src/main/java/org/ladysnake/blabber/impl/common/model/DialogueIllustrationSelectorEntity.java

```diff
@@ -0,0 +1,125 @@
+import org.ladysnake.blabber.api.DialogueIllustrationType;
+
+import java.util.Optional;
+import java.util.UUID;
+
+public record DialogueIllustrationSelectorEntity(Either<String, LivingEntity> selector, int x, int y, int x2, int y2,
+                                                 int size, float yOff, Optional<Integer> stareAtX,
+                                                 Optional<Integer> stareAtY) implements DialogueIllustration {
+    private static final Codec<DialogueIllustrationSelectorEntity> CODEC = RecordCodecBuilder.create(instance -> instance.group(
+            Codec.STRING.fieldOf("entity")
+                    .xmap(Either::<String, LivingEntity>left, e -> e.map(s -> s, Entity::getUuidAsString))
+                    .forGetter(DialogueIllustrationSelectorEntity::selector),
+            Codec.INT.fieldOf("x").forGetter(DialogueIllustrationSelectorEntity::x),
+            Codec.INT.fieldOf("y").forGetter(DialogueIllustrationSelectorEntity::y),
+            Codec.INT.fieldOf("x2").forGetter(DialogueIllustrationSelectorEntity::x2),
+            Codec.INT.fieldOf("y2").forGetter(DialogueIllustrationSelectorEntity::y2),
+            Codec.INT.fieldOf("size").forGetter(DialogueIllustrationSelectorEntity::size),
+            Codecs.createStrictOptionalFieldCodec(Codec.FLOAT, "y_offset", 0.0f).forGetter(DialogueIllustrationSelectorEntity::yOff),
+            Codecs.createStrictOptionalFieldCodec(Codec.INT, "stare_at_x").forGetter(DialogueIllustrationSelectorEntity::stareAtX),
+            Codecs.createStrictOptionalFieldCodec(Codec.INT, "stare_at_y").forGetter(DialogueIllustrationSelectorEntity::stareAtY)
+    ).apply(instance, DialogueIllustrationSelectorEntity::new));
+
+    @Override
+    public void render(DrawContext context, TextRenderer textRenderer, int x, int y, int mouseX, int mouseY, float tickDelta) {
+        selector.ifRight(e -> {
+            int fakedMouseX = stareAtX.map(s -> s + x + this.x).orElse(mouseX);
+            int fakedMouseY = stareAtY.map(s -> s + y + this.y).orElse(mouseY);
+            InventoryScreen.drawEntity(context, this.x + x, this.y + y, x + x2, y + y2, size, yOff, fakedMouseX, fakedMouseY, e);
+        });
+    }
+
+    @Override
+    public void writeToPacket(PacketByteBuf buf) {
+        buf.writeEither(selector, PacketByteBuf::writeString, (b, e) -> b.writeUuid(e.getUuid()));
+        buf.writeInt(x);
+        buf.writeInt(y);
+        buf.writeInt(x2);
+        buf.writeInt(y2);
+        buf.writeInt(size);
+        buf.writeFloat(yOff);
+        buf.writeOptional(stareAtX, PacketByteBuf::writeInt);
+        buf.writeOptional(stareAtY, PacketByteBuf::writeInt);
+    }
+
+    @Override
+    public DialogueIllustrationType<?> getType() {
+        return IllustrationType.INSTANCE;
+    }
+
+    @Override
+    public DialogueIllustrationSelectorEntity parseText(@Nullable ServerCommandSource source, @Nullable Entity sender) throws CommandSyntaxException {
```

Comment:
That's a fairly smart use of `parseText`, makes me wonder if the method should be renamed